lib/service_actor/checkable.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module ServiceActor::Checkable
+  def self.included(base)
+    base.prepend(PrependedMethods)
+  end
+
+  module PrependedMethods
+    def _call
+      checks_for(:input)
+      super
+      checks_for(:output)
+    end
+
+    private
+
+    def checks_for(origin) # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+      self.class.public_send("#{origin}s").each do |input_key, input_options| # rubocop:disable Metrics/BlockLength
+        input_options.each do |check_name, check_conditions| # rubocop:disable Metrics/BlockLength
+          checks = %w[
+            TypeCheck
+            MustCheck
+            InclusionCheck
+            NilCheck
+            DefaultCheck
+          ]
+
+          checks.each do |check_class_name|
+            check_class =
+              Object.const_get("ServiceActor::Checks::#{check_class_name}")
+
+            argument_errors = check_class.check(
+              check_name: check_name,
+              origin: origin.to_sym,
+              input_key: input_key,
+              actor: self.class,
+              value: result[input_key],
+
+              # TypeCheck
+              type_definition: check_conditions,
+              given_type: result[input_key],
```

Comment:
I think the `TypeCheck` should be the one deciding what `value` means for it, and not have it decided here. We can then remove the duplication of ` :result[input_key]` and keep the responsability inside the checks.

Reply:
Done: https://github.com/sunny/actor/pull/97/commits/3ac241044e8334a82aa2549e233237c81e9d15ac